From dd923fcc2151d2d077542870304bec94f094bae7 Mon Sep 17 00:00:00 2001
From: Hans de Goede <johannes.goede@oss.qualcomm.com>
Date: Mon, 25 May 2026 13:47:48 +0200
Subject: [PATCH 02491/15333] arm64: dts: qcom: hamoa: Fix xbl_tmp_buffer_mem
 size

The EFI memtable (shown when booting with efi=debug) shows that the
xbl_tmp_buffer_mem region size is 0x1c0000 bytes large not 0xc00000 bytes:

efi:   0x000082800000-0x0000829bffff [Reserved    |
efi:   0x0000829c0000-0x000083efffff [Conventional|

This fix aligns the size with the EFI reported size and makes /proc/iomem
correctly show mem blocks matching the EFI memmap:

82800000-829bffff : reserved
829c0000-83efffff : System RAM

Instead of:

82800000-833fffff : reserved
83400000-83efffff : System RAM

before this change

Signed-off-by: Hans de Goede <johannes.goede@oss.qualcomm.com>
Link: https://lore.kernel.org/r/20260525114749.5939-2-johannes.goede@oss.qualcomm.com
Signed-off-by: Bjorn Andersson <andersson@kernel.org>
---
 arch/arm64/boot/dts/qcom/hamoa.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/hamoa.dtsi b/arch/arm64/boot/dts/qcom/hamoa.dtsi
index 62126a194576..7f849f5b32d3 100644
--- a/arch/arm64/boot/dts/qcom/hamoa.dtsi
+++ b/arch/arm64/boot/dts/qcom/hamoa.dtsi
@@ -592,7 +592,7 @@ tz_stat_mem: tz-stat@82700000 {
 		};
 
 		xbl_tmp_buffer_mem: xbl-tmp-buffer@82800000 {
-			reg = <0x0 0x82800000 0x0 0xc00000>;
+			reg = <0x0 0x82800000 0x0 0x1c0000>;
 			no-map;
 		};
 
-- 
2.55.0

